extensions/wikia/Flags/specials/SpecialFlagsController.class.php

```diff
@@ -40,6 +40,67 @@ public function index() {
 		$this->hasAdminPermissions = $this->wg->User->isAllowed( 'flags-administration' );
 	}
 
+	public function addFlagType() {
```

Comment:
This method and the `updateFlagType` method below will bypass CSRF protections in the API as the `FlagsApiController::processRequest` method checks if the request is internal, which it now will be due to using `sendRequest`. You can set it as not internal in the `sendRequest` call, but that probably wouldn't account for the POST requirement, so you'd need to duplicate CSRF protections here...